src/asset_manager/asset_manager.rs

```diff
@@ -280,6 +320,101 @@ impl AssetStore for DirectoryStore {
     }
 }
 
+impl AssetLoaderRaw for imagefmt::Image<u8> {
+    fn from_raw(_: &Assets, data: &[u8]) -> Option<imagefmt::Image<u8>> {
+        imagefmt::read_from(&mut Cursor::new(data), imagefmt::ColFmt::RGBA).ok()
+    }
+}
+
+impl AssetLoader<Texture> for imagefmt::Image<u8> {
+    fn from_data(assets: &mut Assets, image: imagefmt::Image<u8>) -> Option<Texture> {
+        let pixels = image.buf.chunks(4).map(|p| [p[0], p[1], p[2], p[3]]).collect::<Vec<_>>();
+        let chunked = pixels.as_slice().chunks(image.w).collect::<Vec<_>>();
```

Comment:
👍 Thanks! I've fixed the texture loading, and added some comments to the `TextureLoadData` struct to help prevent anyone else from making the same mistake